### PR TITLE
chore(litmus-portal): Add Add Probe modal

### PR DESCRIPTION
**Pull request by saswatamcode**
## Proposed changes

This PR adds the Add Probe modal which will be used in the Tune Workflow screens.
### Screenshots
Screenshot 2021-03-03 at 11 52 40 AM
Screenshot 2021-03-03 at 11 52 54 AM

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                })
+              }
+              inputProps={{
+                id: 'probe-type',
+              }}
+            >
+              <MenuItem value="Http">Http</MenuItem>
+              <MenuItem value="cmd">cmd</MenuItem>
+              <MenuItem value="k8s">k8s</MenuItem>
+              <MenuItem value="Prom">Prom</MenuItem>
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:84]: can be used from translations file

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            </Select>
+          </div>
+          <div className={classes.formField}>
+            <InputLabel className={classes.formLabel} htmlFor="probe-prop">
+              Probe Properties
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:89]: add translation

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+              }
+              inputProps={{
+                id: 'probe-prop',
+              }}
+            >
+              <MenuItem value="SoT">SoT</MenuItem>
+              <MenuItem value="EoT">EoT</MenuItem>
+              <MenuItem value="Edge">Edge</MenuItem>
+              <MenuItem value="Continous">Continuos</MenuItem>
+              <MenuItem value="onChaos">onChaos</MenuItem>
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:109]: translations can be added

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            </Select>
+          </div>
+          <hr style={{ width: '100%' }} />
+
+          <div className={classes.subHeading}>Probe Properties</div>
+
+          <div style={{ display: 'flex', width: '100%' }}>
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:116]: use from styles, avoid inline styling

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+            <div className={classes.formField}>
+              <InputLabel className={classes.formLabel} htmlFor="timeout">
+                Timeout
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:119]: translation

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]
@saswatamcode please add all the translations

**Comment by ishangupta-ds** [comment]
Screenshot 2021-03-03 at 4 54 21 PM
Screenshot 2021-03-03 at 4 54 44 PM
Screenshot 2021-03-03 at 4 55 07 PM

@saswatamcode PTAL

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+                setProbeData({
+                  ...probeData,
+                  probeType: e.target.value as string,
+                })
+              }
+              inputProps={{
+                id: 'probe-type',
+              }}
+            >
+              <MenuItem value="Http">
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:86]: Can we make them uniform either all pascal or lower case? (preferred lower case)
{ Http, Cmd, K8s, Prom} OR {http, cmd, k8s, prom}

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+                {t('createWorkflow.tuneWorkflow.addProbe.select.k8s')}
+              </MenuItem>
+              <MenuItem value="Prom">
+                {t('createWorkflow.tuneWorkflow.addProbe.select.prom')}
+              </MenuItem>
+            </Select>
+          </div>
+          <div className={classes.formField}>
+            <InputLabel className={classes.formLabel} htmlFor="probe-prop">
+              {t('createWorkflow.tuneWorkflow.addProbe.labels.probeProp')}
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:102]: SOT, EOT, Edge,  Continuous, OnChaos all comes under `mode`. refer: https://docs.litmuschaos.io/docs/litmus-probe/

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+                setProbeData({
+                  ...probeData,
+                  probeProperties: e.target.value as string,
+                })
+              }
+              inputProps={{
+                id: 'probe-prop',
+              }}
+            >
+              <MenuItem value="SoT">
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:118]: SOT

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+                {t('createWorkflow.tuneWorkflow.addProbe.select.sot')}
+              </MenuItem>
+              <MenuItem value="EoT">
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:121]: EOT

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+                {t('createWorkflow.tuneWorkflow.addProbe.select.eot')}
+              </MenuItem>
+              <MenuItem value="Edge">
+                {t('createWorkflow.tuneWorkflow.addProbe.select.edge')}
+              </MenuItem>
+              <MenuItem value="Continous">
+                {t('createWorkflow.tuneWorkflow.addProbe.select.continuos')}
+              </MenuItem>
+              <MenuItem value="onChaos">
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:130]: OnChaos

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+                    ...probeData,
+                    pollingInterval: e.target.value,
+                  })
+                }
+              />
+            </div>
+          </div>
+          <div className={classes.formField}>
+            <InputLabel className={classes.formLabel} htmlFor="interval-delay">
+              {t('createWorkflow.tuneWorkflow.addProbe.labels.intervalDelay')}
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:208]: initialDelaySeconds

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+          <div className={classes.formField}>
+            <InputLabel className={classes.formLabel} htmlFor="url">
+              {t('createWorkflow.tuneWorkflow.addProbe.labels.url')}
+            </InputLabel>
+            <InputField
+              variant="primary"
+              width="50%"
+              id="url"
+              type="text"
+              value={probeData.url}
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:237]: Is this static or dynamic (changed based on probeType)? The corresponding input options will come based on the selected probe-type?

File: litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+          <div className={classes.formField}>
+            <InputLabel className={classes.formLabel} htmlFor="response">
+              {t('createWorkflow.tuneWorkflow.addProbe.labels.expectedRes')}
+            </InputLabel>
+            <InputField
+              variant="primary"
+              width="50%"
+              id="response"
+              type="number"
+              value={probeData.expectedResponseCode}
```
Comment (ispeakc0de) [inline litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/AddProbe/index.tsx:252]: The input schema of http probe has been changed, we don't this field now. Please refer: https://docs.litmuschaos.io/docs/litmus-probe/#httpprobe

**Comment by S-ayanide** [comment]
@saswatamcode Please align the key-value pairs with a proper design and not just side by side. The Figma design would need updates too. Put the input fields and labels in proper spacing and alignment, 50% or so should be good. Also please check the DeepScan issues.

**Comment by saswatamcode** [comment]
PR Updated to make inputs dynamic based on Probe Type and for style fixes.
Screenshot 2021-03-04 at 7 51 03 PM
## Probe inputs when type is `http`
Screenshot 2021-03-04 at 7 51 35 PM
## Probe inputs when type is `cmd`
Screenshot 2021-03-04 at 7 51 51 PM
## Probe inputs when type is `k8s`
Screenshot 2021-03-04 at 7 52 07 PM
## Probe inputs when type is `prom`
Screenshot 2021-03-04 at 7 52 24 PM

**Review by ishangupta-ds — APPROVED** [review]
LGTM

**Review by S-ayanide — APPROVED** [review]

**Pull request merged**